src/Operation/Aggregate.php

```diff
@@ -322,7 +322,7 @@ private function createCommandDocument(): array
             'pipeline' => $this->pipeline,
         ];
 
-        foreach (['allowDiskUse', 'bypassDocumentValidation', 'comment', 'explain', 'maxTimeMS'] as $option) {
+        foreach (['allowDiskUse', 'bypassDocumentValidation', 'comment', 'explain', 'maxTimeMS', 'readConcern'] as $option) {
```

Comment:
`readConcern` should _not_ be added directly to the command document. The option is already [handled in `executeCommand()`](https://github.com/mongodb/mongo-php-library/blob/1.15.0/src/Operation/Aggregate.php#L372) along with other options for the execute method.

That code path allows for sensible inheritance of client-level options. I'd have to double-check libmongoc internals to confirm, but I think directly adding it in a command document here might result in duplicate BSON fields being sent to the server. I know that's possible if you were to manually inject a session ID in the `lsid` field.

Instead, I think you should add this option in `getCommandDocument()`. You can use the array union operator to do so and just add `['readConcern' => ...]` to `$this->createCommandDocument()` before returning. A more verbose approach with an assigned variable is fine as well.

----

Likewise for other Explainable commands where you modified the value used to construct the Command object instead of the `Explainable::getCommandDocument()` implementation.

Reply:
I understand better how it works now.
I opted for the more verbose way to add the `readConcern` option when available: its easier to read and [better for code analysis](https://localheinz.com/articles/2023/03/18/avoiding-one-liners-in-php/).